File: src/main/java/guru/xgm/language/jass/extapi/psi/JassPsiFileBase.java
package guru.xgm.language.jass.extapi.psi;

import guru.xgm.language.jass.lang.JassLanguage;
import guru.xgm.language.jass.openapi.fileTypes.JassFileType;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import org.jetbrains.annotations.NotNull;

public class JassPsiFileBase extends PsiFileBase {

    public JassPsiFileBase(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, JassLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return JassFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "JASS";
    }
}

File: src/main/java/guru/xgm/language/jass/extapi/psi/JassPsiFileBase.kt
package guru.xgm.language.jass.extapi.psi

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider
import guru.xgm.language.jass.lang.JassLanguage
import guru.xgm.language.jass.openapi.fileTypes.JassFileType

class JassPsiFileBase(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, JassLanguage.instance) {
    override fun getFileType(): FileType {
        return JassFileType.INSTANCE
    }

    override fun toString(): String {
        return JassLanguage.NAME
    }
}
